alf/layers.py

```diff
@@ -11,104 +11,179 @@
+            out_channels (int): channels of the output image
+            kernel_size (int):
+            activation (torch.nn.functional):
+            strides (int):
+            padding (int):
+            use_bias (bool):
+            kernel_init_gain (float): a scaling factor (gain) applied to the
+                std of kernel init distribution
+            bias_init_value (float): a constant
+        """
+        super(Conv2D, self).__init__()
+        self._activation = activation
+        self._conv2d = nn.Conv2d(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=strides,
+            padding=padding,
+            bias=use_bias)
+        variance_scaling_init(
+            self._conv2d.weight.data,
+            gain=kernel_init_gain,
+            nonlinearity=self._activation.__name__)
+        if use_bias:
+            nn.init.constant_(self._conv2d.bias.data, bias_init_value)
+
+    def forward(self, img):
+        return self._activation(self._conv2d(img))
+
+    @property
+    def weight(self):
+        return self._conv2d.weight
+
+    @property
+    def bias(self):
+        return self._conv2d.bias
 
 
 @gin.configurable
-def get_identity_layer():
-    return tf.keras.layers.Lambda(lambda x: x)
+class ConvTranspose2D(nn.Module):
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size,
+                 activation=torch.relu,
+                 strides=1,
+                 padding=0,
+                 use_bias=True,
+                 kernel_init_gain=1.0,
+                 bias_init_value=0.0):
+        """A 2D ConvTranspose layer that's also responsible for activation and
+        customized weights initialization. An auto gain calculation might depend
+        on the activation following the conv layer. Suggest using this wrapper
+        module instead of nn.ConvTranspose2d if you really care about weight std
+        after init.
 
-
-def get_first_element_layer():
-    return tf.keras.layers.Lambda(lambda x: x[0])
+        Args:
+            in_channels (int): channels of the input image
+            out_channels (int): channels of the output image
+            kernel_size (int):
+            activation (torch.nn.functional):
+            strides (int):
+            padding (int):
+            use_bias (bool):
+            kernel_init_gain (float): a scaling factor (gain) applied to the
+                std of kernel init distribution
+            bias_init_value (float): a constant
+        """
+        super(ConvTranspose2D, self).__init__()
+        self._activation = activation
+        self._conv_trans2d = nn.ConvTranspose2d(
+            in_channels,
+            out_channels,
+            kernel_size,
+            stride=strides,
+            padding=padding,
+            bias=use_bias)
+        variance_scaling_init(
+            self._conv_trans2d.weight.data,
```

Comment:
is weight.data transposed or not? If it is transposed, then the fanin fanout need to be switched.

Reply:
It's already transposed

Reply:
Yeah it's transposed.